tests/stub/driver_execute_query/test_driver_execute_query.py

```diff
@@ -0,0 +1,262 @@
+        self._start_server(self._router, "router_with_db_name.script")
+        self._start_server(self._writer, "tx_return_1_with_params.script")
+        self._driver = self._new_driver()
+
+        eager_result = self._driver.execute_query("RETURN 1 AS n", {
+            "a": types.CypherInt(1)
+        }, {"database": "neo4j"})
+
+        self.assertEqual(eager_result.keys, ["n"])
+        self.assertEqual(eager_result.records, [
+                         types.Record(values=[types.CypherInt(1)])])
+        self.assertIsNotNone(eager_result.summary)
+
+    def test_configure_impersonated_user(self):
+        self._start_server(self._router, "router_with_impersonation.script")
+        self._start_server(
+            self._writer, "tx_return_1_with_impersonation.script")
+        self._driver = self._new_driver()
+
+        eager_result = self._driver.execute_query(
+            "RETURN 1 AS n",
+            config={"impersonatedUser": "that-other-dude"})
+
+        self.assertEqual(eager_result.keys, ["n"])
+        self.assertEqual(eager_result.records, [
+                         types.Record(values=[types.CypherInt(1)])])
+        self.assertIsNotNone(eager_result.summary)
+
+    def test_causal_consistency_between_query_executions(self):
+        self._start_server(self._router, "router.script")
+        self._start_server(
+            self._writer, "transaction_chaining.script")
+        self._driver = self._new_driver()
+
+        # CREATING NODE
+        eager_result = self._driver.execute_query(
+            "CREATE (p:Person{name:$name}) RETURN p.name AS name", {
+                "name": types.CypherString("the person")
+            })
+
+        self.assertEqual(eager_result.keys, ["name"])
+        self.assertEqual(eager_result.records, [
+                         types.Record(values=[
+                             types.CypherString("the person")])])
+        self.assertIsNotNone(eager_result.summary)
+
+        # READING SAME NODE
+        eager_result2 = self._driver.execute_query(
+            "MATCH (p:Person{name:$name}) RETURN p.name AS name", {
+                "name": types.CypherString("the person")
+            })
+
+        self.assertEqual(eager_result2.keys, ["name"])
+        self.assertEqual(eager_result2.records, [
+                         types.Record(values=[
+                             types.CypherString("the person")])])
+        self.assertIsNotNone(eager_result2.summary)
+
+    def test_disable_bookmark_manager(self):
+        self._start_server(self._router, "router.script")
+        self._start_server(
+            self._writer, "transaction_chaining.script")
+        self._driver = self._new_driver()
+
+        # CREATING NODE
+        eager_result = self._driver.execute_query(
+            "CREATE (p:Person{name:$name}) RETURN p.name AS name", {
+                "name": types.CypherString("the person")
+            }, {
+                "bookmarkManagerId": None
```

Comment:
Note that some languages have a difficult time differentiating a missing key from a key with a `null` value in a JSON object.

However, I'm not sure we have to do something about it as there are other parts of the TestKit protocol that also require the backends to be sensitive to this.

Another thing regarding this: I'm not a huge fan of magic strings in the tests. So I find the fact that the tests need to know that this config option is called `"bookmarkManagerId"` not ideal. If you want to, I can propose an alternative design.

Reply:
I was thinking about using `**kwargs` instead and let the translation to the protocol message, specially because the "bookmarkManagerId" has difference between not bet set or be set to None. Maybe we could work around using `-1` for when it is `not set`

Reply:
`-1`: "use no BMM", `null`: "use default BMM" sounds like a good idea to me.

Instead of using `**kwargs`, you could use something like

```python
def execute_query(self, cypher, params=None, database=None, other_config_option_1=None,  other_config_option_2=None, bookmark_manager=...):
    config = {}
    if bookmark_manager is not ...:
        if bookmark_manager is None:
            config["bookmarkManagerId"] = None
        else:
            config["bookmarkManagerId"] = bookmark_manager.id
   ...     
```

Reply:
Sure, I will address it.